bokeh/tile_providers.py

```diff
@@ -1,59 +1,109 @@
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # Copyright (c) 2012 - 2019, Anaconda, Inc., and Bokeh Contributors.
 # All rights reserved.
 #
 # The full license is in the file LICENSE.txt, distributed with this software.
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 ''' Pre-configured tile sources for common third party tile services.
 
 
 Attributes:
 
+    Provider
+        Enum listing all available tile services
+
+    get_provider
+        Use this function to retrieve any tile provider.
+
+        Args:
+            provider_name (Union[str, Provider])
+                Name of the tile provider to supply.
+                Use tile_providers.Provider enum, or the name of a provider as string
+
+        Returns:
+            WMTSTileProviderSource: The desired tile provider instance
+
+        Raises:
+            ValueError, if the provider can not be found
+
+
+        Example:
+
+        .. code-block:: python
+
+                >>> from bokeh.tile_providers import get_provider, Provider
+                >>> get_provider(Provider.CARTODBPOSITRON)
+                <class 'bokeh.models.tiles.WMTSTileSource'>
+                >>> get_provider('CARTODBPOSITRON')
+                <class 'bokeh.models.tiles.WMTSTileSource'>
+
+
     CARTODBPOSITRON
         Tile Source for CartoDB Tile Service
 
+        Warns:
+            BokehDeprecationWarning: Deprecated in bokeh 2.0.0. Use get_provider instead
```

Comment:
The deprecation will be executed when 2.0 happens (i.e. that's when breaking changes will happen) but the deprecation will start from 1.1 (presumably) so we should state that throughout.